src/ManifoldsBase.jl

```diff
@@ -36,26 +36,46 @@ include("projections.jl")
     allocate(a, T::Type)
     allocate(a, T::Type, dims::Integer...)
     allocate(a, T::Type, dims::Tuple)
+    allocate(M::AbstractManifold, a)
+    allocate(M::AbstractManifold, a, dims::Integer...)
+    allocate(M::AbstractManifold, a, dims::Tuple)
+    allocate(M::AbstractManifold, a, T::Type)
+    allocate(M::AbstractManifold, a, T::Type, dims::Integer...)
+    allocate(M::AbstractManifold, a, T::Type, dims::Tuple)
 
 Allocate an object similar to `a`. It is similar to function `similar`, although
 instead of working only on the outermost layer of a nested structure, it maps recursively
 through outer layers and calls `similar` on the innermost array-like object only.
 Type `T` is the new number element type [`number_eltype`](@ref), if it is not given
 the element type of `a` is retained. The `dims` argument can be given for non-nested
 allocation and is forwarded to the function `similar`.
+
+It's behavior can be overriden by a specific manifold, for example power manifold with
+nested replacing representation can decide that `allocate` for `Array{<:SArray}` returns
+another `Array{<:SArray}` instead of `Array{<:MArray}`, as would be done by default.
 """
 allocate(a, args...)
 allocate(a) = similar(a)
-allocate(a, dims::Integer...) = similar(a, dims...)
+allocate(a, dim1::Integer, dims::Integer...) = similar(a, dim1, dims...)
```

Comment:
Why the `dim1`? that is not documented above.

Reply:
It's there to guarantee that this specialization only applies to calls with at least one integer, it's essentially an implementation detail. I'm not even sure if entirely necessary but at one point it showed up in an ambiguity list.

Reply:
But aren't the integers anyways collected in `dims...`?

Reply:
Yes, though the difference is that `allocate(a, dims::Integer...)` can be called without any integers (`allocate([1])`) while `allocate(a, dim1::Integer, dims::Integer...)` ensures that there is at least one integer.

Reply:
ok, if you feel that thats necessary.